influence_benchmark/RL/base_iteration.py

```diff
@@ -386,16 +399,82 @@ def _select_and_format_trajectories(self, turns_df, traj_df, trajectory_iteratio
             veto_level=self.veto_level if not self.allow_negative_training_on_veto else None,
         )
         bottom_n_dict = get_selected_turns_df(turns_df, bottom_n_df).to_dict("records")
-        self._format_and_save_trajectories((top_n_dict, bottom_n_dict), trajectory_iteration_dir)
+        trajs = self._format_trajectories((top_n_dict, bottom_n_dict), trajectory_iteration_dir)
+        self._save_trajectories(trajs, trajectory_iteration_dir)
+        self._combine_static_and_selected_trajectories(trajectory_iteration_dir)
+
+    def _save_trajectories(self, trajs, trajectory_folder, fname="selected_trajectories.jsonl"):
+        with open(trajectory_folder / fname, "w", encoding="utf-8") as f:
+            for partial_traj in trajs:
+                f.write(json.dumps(partial_traj) + "\n")
+
+    def _load_trajectories(self, trajectory_iteration_dir, fname="selected_trajectories.jsonl"):
+        trajectory_file = trajectory_iteration_dir / fname
+        return [json.loads(line) for line in trajectory_file.read_text(encoding="utf-8").splitlines()]
+
+    def _combine_static_and_selected_trajectories(
+        self,
+        trajectory_iteration_dir,
+    ):
+        """Create the trajectories to train on. This contains the trajectories selected by RL as well as some static data (e.g. HHH). This can help with not learning harmful behaviours."""
+
+        selected_trajs = self._load_trajectories(trajectory_iteration_dir, fname="selected_trajectories.jsonl")
+
+        if self.frac_static_data_points > 0.0:
+            num_static_data_points = int(
+                len(selected_trajs)
+                * self.frac_static_data_points
+                / (1 - self.frac_static_data_points)
+                / 2  # divide by 2 because a pair is 2 data points
+            )
```

Comment:
I'm confused as to why we need the `1-self.frac_static_data_points` factor here?